app/main.py

```diff
@@ -1,3 +1,29 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.additional_funcs import (create_instances,
+                                  total_trip_cost,
+                                  customer_can_go_to_shop,
+                                  print_shop_receipt,
+                                  taking_back_home)
+
+
+def shop_trip() -> None:
+    with open("config.json", "r") as c:
+        config = json.load(c)
+    fuel_price = config["FUEL_PRICE"]
+    customers = config["customers"]
+    shops = config["shops"]
+    customers_list, shops_list = create_instances(customers, shops)
+    total_trip_cost(fuel_price, customers_list, shops_list)
+    for customer in customers_list:
+        trip_to_shop = (customer_can_go_to_shop(customer))
```

Comment:
The variable `trip_to_shop` might be `None` if `customer_can_go_to_shop` returns `None` when the customer cannot afford any trip. Ensure that this case is handled to avoid potential errors.